File: src/types/api/marine.ts
export type DivingSpot = {
    name: string
    lat: number
    lon: number
}

export type MarineCurrent = {
    time: string
    interval: number
    wave_height: number | null
    wind_speed_10m: number | null
    sea_surface_temperature: number | null
}

export type MarineResponse = {
    latitude: number
    longitude: number
    generationtime_ms: number
    utc_offset_seconds: number
    timezone: string
    timezone_abbreviation: string
    elevation: number
    current_units: {
        time: string
        interval: string
        wave_height: string
        wind_speed_10m: string
        sea_surface_temperature: string
    }
    current: MarineCurrent
}

File: backend/app/schemas/marine.py
from pydantic import BaseModel


class DivingSpot(BaseModel):
    name: str
    lat: float
    lon: float


class PrefectureInfo(BaseModel):
    code: str
    name: str


class CurrentUnits(BaseModel):
    time: str
    interval: str
    wave_height: str
    wind_speed_10m: str
    sea_surface_temperature: str


class Current(BaseModel):
    time: str
    interval: int
    wave_height: float | None
    wind_speed_10m: float | None
    sea_surface_temperature: float | None


class MarineResponse(BaseModel):
    latitude: float
    longitude: float
    generationtime_ms: float
    utc_offset_seconds: int
    timezone: str
    timezone_abbreviation: str
    elevation: float
    current_units: CurrentUnits
    current: Current